CondCore/SiStripPlugins/interface/SiStripPayloadInspectorHelper.h

```diff
@@ -3,29 +3,223 @@
+    case SiStripPI::TID1r: return "TID D1 r-#varphi";
+    case SiStripPI::TID1s: return "TID D1 stereo";
+    case SiStripPI::TID2r: return "TID D2 r-#varphi";
+    case SiStripPI::TID2s: return "TID D2 stereo";
+    case SiStripPI::TID3r: return "TID D3 r-#varphi"; 
+    case SiStripPI::TID3s: return "TID D3 stereo";
+    case SiStripPI::END_OF_REGIONS : return "undefined";
+    default : return "should never be here";  
+    }
+  }
 
-  return std::make_pair(m-2*stdev,m+2*stdev);
+  /*--------------------------------------------------------------------*/
+  std::pair<float,float> getTheRange(std::map<uint32_t,float> values)
+  /*--------------------------------------------------------------------*/
+  {
+    float sum = std::accumulate(std::begin(values), 
+				std::end(values), 
+				0.0,
+				[] (float value, const std::map<uint32_t,float>::value_type& p)
+				{ return value + p.second; }
+				);
+    
+    float m =  sum / values.size();
+    
+    float accum = 0.0;
+    std::for_each (std::begin(values), 
+		   std::end(values), 
+		   [&](const std::map<uint32_t,float>::value_type& p) 
+		   {accum += (p.second - m) * (p.second - m);}
+		   );
+    
+    float stdev = sqrt(accum / (values.size()-1)); 
+    
+    return std::make_pair(m-2*stdev,m+2*stdev);
+    
+  }
+  
+  /*--------------------------------------------------------------------*/
+  void drawStatBox(std::map<std::string,std::shared_ptr<TH1F>> histos, std::map<std::string,int> colormap, std::vector<std::string> legend, double X=0.15, double Y=0.93, double W=0.15, double H=0.10)
+  /*--------------------------------------------------------------------*/
+  {  
+    char   buffer[255];
+    
+    int i=0;
+    for ( const auto &element : legend ){
+      TPaveText* stat = new TPaveText(X,Y-(i*H), X+W, Y-(i+1)*H, "NDC");
+      i++;
+      auto Histo = histos[element];
+      sprintf(buffer,"Entries : %i\n",(int)Histo->GetEntries());
+      stat->AddText(buffer);
+      
+      sprintf(buffer,"Mean    : %6.2f\n",Histo->GetMean());
+      stat->AddText(buffer);
+      
+      sprintf(buffer,"RMS     : %6.2f\n",Histo->GetRMS());
+      stat->AddText(buffer);
+      
+      stat->SetFillColor(0);
+      stat->SetLineColor(colormap[element]);
+      stat->SetTextColor(colormap[element]);
+      stat->SetTextSize(0.03);
+      stat->SetBorderSize(0);
+      stat->SetMargin(0.05);
+      stat->SetTextAlign(12);
+      stat->Draw();
+    }
+  }
+  
+  /*--------------------------------------------------------------------*/
+  std::pair<float,float> getExtrema(TH1 *h1,TH1 *h2)
+  /*--------------------------------------------------------------------*/
+  {
+    float theMax(-9999.);
+    float theMin(9999.);
+    theMax =  h1->GetMaximum() > h2->GetMaximum() ?  h1->GetMaximum() :  h2->GetMaximum();
+    theMin =  h1->GetMinimum() < h2->GetMaximum() ?  h1->GetMinimum() :  h2->GetMinimum();
+    
+    auto result = std::make_pair(theMin*0.95,theMax*1.05);
```

Comment:
hi @mmusich - these are appropriate only for positive max/min values. Is that the only scenario for these functions?